### PR TITLE
Updates required to work on Django 1.7

### PR DESCRIPTION
**Pull request by robhudson**

### CODE REVIEW

File: mkt/settings.py [hunk 1]
```diff
@@ -1001,8 +1001,13 @@ def JINJA_CONFIG():
 READ_ONLY = False
 
 REDIS_BACKENDS = {
```
Comment (diox) [inline mkt/settings.py]: `REDIS_BACKENDS` is also defined in `sites/*`, and they use `private` settings. I assume thos are not using the dict format either and will fail with the update...

Comment (robhudson) [inline mkt/settings.py]: Yep, I need to come back to this. I think I will copy the single string parser that Django 1.7 removed into our codebase.

Comment (diox) [inline mkt/settings.py]: Does the dict-like syntax works now with 1.6 ? If it does, let's convert everything now and then we don't have to have the single string parser ?

Comment (robhudson) [inline mkt/settings.py]: The nice thing about the single string parse is you can easily override this with an environment variable, which is how it is currently on master for docker. It's something I broke here and need to fix.

File: manage.py [hunk 2]
```diff
@@ -30,16 +30,6 @@
...
 log_configure()
 
-# We need to import waffle here to avoid a circular import with jingo which
-# loads all INSTALLED_APPS looking for helpers.py files, but some of those apps
-# import jingo.
-import waffle  # flake8: noqa
-
-# Hardcore monkeypatching action.
-import jingo.monkey
-jingo.monkey.patch()
```
Comment (andymckay) [inline manage.py]: i'm glad to see this go, we should check receiptcheck still works though

File: mkt/settings.py (hunk 1 above)
Comment (andymckay) [inline mkt/settings.py]: For docker, all we need is the host, never seen any of other stuff change. Just setting `REDIS_HOST` as an environment variable would work for me.

File: mkt/developers/views.py [hunk 3]
```diff
@@ -553,6 +553,7 @@ def ownership(request, addon_id, addon):
                     thread___addon=addon).delete()
 
         for author in user_form.deleted_objects:
+            author.delete()
```
Comment (andymckay) [inline mkt/developers/views.py]: why?

File: mkt/urls.py [hunk 4]
```diff
@@ -18,6 +18,11 @@
 from mkt.users.views import logout
 
 
+# Hardcore monkeypatching action.
+import jingo.monkey
+jingo.monkey.patch()
```
Comment (andymckay) [inline mkt/urls.py]: will we need to move the import waffle here? thinking of receiptcheck primarily.

File: mkt/developers/views.py (hunk 3 above)
Comment (robhudson) [inline mkt/developers/views.py]: That may be sloppy. I'll have to look at that again. We don't want to delete the UserProfile object, but the object that associates a user with an admin.

The reason I believe is this:
https://docs.djangoproject.com/en/dev/releases/1.7/#miscellaneous (bullet point 2)

File: mkt/settings.py (hunk 1 above)
Comment (robhudson) [inline mkt/settings.py]: Ok, cool, thanks @andymckay

File: mkt/urls.py (hunk 4 above)
Comment (robhudson) [inline mkt/urls.py]: I'm not sure, we'd have to test. When I did this it worked w/o the extra import but I didn't check receiptcheck.

File: mkt/developers/views.py (hunk 3 above)
Comment (andymckay) [inline mkt/developers/views.py]: Oh this is the AddonUser model that's being deleted. That should be ok, I was worried about deleting something that cascaded with terrible consequences.

Comment (robhudson) [inline mkt/developers/views.py]: I'd like to verify that but yes. I should name the variable `app_author` or something instead.

**Comment by robhudson** [comment]
Rebased. I'm going to open this as ready for review.

File: mkt/settings.py [hunk 5]
```diff
@@ -959,7 +959,7 @@ def JINJA_CONFIG():
     # Stops nose always adding lib into the path, which then breaks crypto
     # imports.
     '-P',
-    '--with-fixture-bundling',
+#    '--with-fixture-bundling',
```
Comment (diox) [inline mkt/settings.py]: Why ? That's probably bad for test perf.

File: mkt/settings.py [hunk 6]
```diff
@@ -1006,8 +1006,13 @@ def JINJA_CONFIG():
 READ_ONLY = False
 
 REDIS_BACKENDS = {
-    'master': os.environ.get('REDIS_URL',
```
Comment (diox) [inline mkt/settings.py]: I don't remember what was the conclusion of the discussions on this ; don't you need to patch the settings in `sites/` too ?

**Comment by diox** [comment]
Tests pass but with `django.db.utils.OperationalError: (2006, 'MySQL server has gone away')` at the end so they are marked as failed...

**Comment by robhudson** [comment]
Yeah, and I haven't figured out why. But I haven't looked too hard yet.

**Comment by robhudson** [comment]
Closing PR to move branch to the mozilla origin.

**Pull request closed without merging**